net/tcp/tcp_send_buffered.c

```diff
@@ -376,6 +376,29 @@ static uint16_t psock_send_eventhandler(FAR struct net_driver_s *dev,
   conn = psock->s_conn;
   DEBUGASSERT(conn != NULL);
 
+#ifdef CONFIG_DEBUG_NET_ERROR
+  if (conn->dev == NULL || (pvconn != conn && pvconn != NULL))
+    {
+      nerr("ERROR: conn->dev == NULL or pvconn != conn:"
+           " dev=%p pvconn=%p pvpriv=%p flags=0x%04x"
```

Comment:
create the new file named tcp_dump.c and unify the tcp connect dump ?

Reply:
The dump of connect could be implement to a common feature, the developers can reuse this api to debug other stack issues

if this PR is a common feature, welcome.
If this PR is just for testing of debugging, please mark the tag as draft and send the patch link to reporter locally, I hope the issue can be resolved before mainline.

about #5252, I will upload relevant patch asap.

Reply:
> The dump of connect could be implement to a common feature, the developers can reuse this api to debug other stack issues

Good. I have created tcp_event_handler_dump() function.